spec/index.bs

```diff
@@ -2959,7 +2967,17 @@ The {{GPUBufferUsage}} flags determine how a {{GPUBuffer}} may be used after its
 
                     1. Set |b|.{{GPUBuffer/[[deviceTimelineState]]}} to "[=buffer device timeline state/available=]".
 
-                1. Set each byte of |b|'s allocation to zero.
+                1. If |mapOutOfMemory| is `true`,
+                    [$Generate an internal error$] with reason `undefined`
+                    and make |b| [=invalid=].
+
+                    Issue: Should the reason be `undefined` or `"out-of-memory"`?
```

Comment:
If you get an OOM from a buffer, you're going to try to free up other buffers to make room for it. As such reporting OOM here would likely be misleading. Which is kind of sad, because you probably COULD resolve it by freeing up JS memory, but we have no way of telling you that.

On the other hand, since you're waiting on the GC you don't really know when the JS memory has actually been freed, since that would involve observing the Garbage Collection, so you won't know when you can safely try this again.

On the other other hand, this is a reasonably compelling reason for differentiating between "internal" and "internal but really OOM", so maybe that helps resolve Myles concerns?